modules/node/src/redis/redis.provider.ts

```diff
@@ -0,0 +1,51 @@
+import { FactoryProvider } from "@nestjs/common/interfaces";
+import Redis from "ioredis";
+import Redlock from "redlock";
+
+import { ConfigService } from "../config/config.service";
+import { RedisProviderId, RedlockProviderId } from "../constants";
+
+export const redisClientFactory: FactoryProvider = {
+  inject: [ConfigService],
+  provide: RedisProviderId,
+  useFactory: (config: ConfigService): Redis.Redis => {
+    const redisClient = new Redis(config.getRedisUrl(), {
+      retryStrategy: (times: number): number => {
+        console.log("Lost connection to redis. Retrying to connect...");
+        const delay = Math.min(times * 50, 2000);
+        return delay;
+      },
+    });
+    return redisClient;
+  },
+};
+
+export const redlockClientFactory: FactoryProvider = {
```

Comment:
This seems like it should be part of the lock module. Is a separate redis module even needed or is this here just in case we need redis somewhere else for something (YAGNI?)

Reply:
Yes, indeed it is here just so the Redis provider itself is reusable :).